src/main/java/duke/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,38 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.main.Storage;
+import duke.main.TaskList;
+import duke.main.Ui;
+import duke.task.Deadline;
+
+import java.util.Date;
+
+public class AddDeadlineCommand extends Command {
+    private String description;
+    private String deadlineBy;
+    private Date deadlineDate;
```

Comment:
Between `deadlineDate` and `deadlineBy`, it seems like one of them will always be null. Maybe you should consider just using a single variable?